addons/portal/models/mail_thread.py

```diff
@@ -30,12 +31,14 @@ def _notify_get_recipients_groups(self, message, model_description, msg_vals=Non
 
         customer = self._mail_get_partners(introspect_fields=False)[self.id]
         if customer:
-            access_token = self._portal_ensure_token()
+            # sudo to access to the partner in case of a portal user with access write on the thread
```

Comment:
You're right. The comment here doesn't make sense and I can't reproduce the use case. I'm sure I had a crash for this line earlier but maybe it's gone after the changes on the PR. I'm gonna double check it and if I can't find it, I remove it.

Reply:
I guess it might be from when we checked `read` access rather than `write` for posting on some models?

I'm pretty sure the issue could still happen if you can find a `read` access thread, with previously no token existing, and sending from a portal user who has `read` but no `write` access, with other customers that have no access and need a token.

I think the `sudo` is still necessary, but the comment should be about `read` and not `write`.

`# sudo: mail.thread - portal user posting with read access should be able to create token when notifying other customers that need it`